sdk/src/wallet/operations/reissue.rs

```diff
@@ -16,48 +16,69 @@ use crate::{
-    /// included (referenced by a milestone). Returns the included block id.
-    pub async fn reissue_transaction_until_included(
+    /// Checks the transaction state for a provided transaction id until it's accepted. Interval in milliseconds.
+    /// Returns the block id that contains this transaction.
+    pub async fn await_transaction_acceptance(
         &self,
         transaction_id: &TransactionId,
         interval: Option<u64>,
         max_attempts: Option<u64>,
     ) -> crate::wallet::Result<BlockId> {
-        log::debug!("[reissue_transaction_until_included]");
+        log::debug!("[await_transaction_acceptance]");
 
-        let protocol_parameters = self.client().get_protocol_parameters().await?;
-        let transaction = self.data().await.transactions.get(transaction_id).cloned();
+        let transaction = self
+            .data()
+            .await
+            .transactions
+            .get(transaction_id)
+            .cloned()
+            .ok_or_else(|| Error::TransactionNotFound(*transaction_id))?;
 
-        if let Some(transaction) = transaction {
-            if transaction.inclusion_state == InclusionState::Confirmed {
-                return transaction.block_id.ok_or(Error::MissingParameter("block id"));
-            }
+        if transaction.inclusion_state == InclusionState::Accepted
+            || transaction.inclusion_state == InclusionState::Confirmed
+            || transaction.inclusion_state == InclusionState::Finalized
+        {
+            return transaction.block_id.ok_or(Error::MissingParameter("block id"));
+        }
 
-            if transaction.inclusion_state == InclusionState::Conflicting
-                || transaction.inclusion_state == InclusionState::UnknownPruned
-            {
-                return Err(ClientError::TangleInclusion(format!(
-                    "transaction id: {} inclusion state: {:?}",
-                    transaction_id, transaction.inclusion_state
-                ))
-                .into());
-            }
+        if transaction.inclusion_state == InclusionState::Conflicting
+            || transaction.inclusion_state == InclusionState::UnknownPruned
+        {
+            return Err(ClientError::TransactionAcceptance(format!(
+                "transaction id: {} inclusion state: {:?}",
+                transaction_id, transaction.inclusion_state
+            ))
+            .into());
+        }
+
+        let block_id = match transaction.block_id {
+            Some(block_id) => block_id,
+            None => {
+                let wallet_data = self.data().await;
+                let issuer_id = transaction
+                    .payload
+                    .transaction()
+                    .allotments()
+                    .first()
+                    .map(|a| *a.account_id())
+                    .or_else(|| wallet_data.first_account_id())
+                    .ok_or(Error::AccountNotFound)?;
```

Comment:
This feels very iffy to me. Maybe we should pass the id in instead?

Reply:
But if it's a different one than allotted it could burn the mana from the wrong account

Reply:
but there's no guarantee that this is right either :c

Reply:
You said we have some issue with the reissuing part in syncing (because of the secret manager or something), don't we have the same issue here?
Because then I maybe just remove this part here?

Reply:
Removed it

Reply:
well that issue is probably not an issue here, but ok